### ISSUE TITLE
Task: Performance of drag and drop of Building blocks.

### ISSUE THREAD

**Issue by rahulbarwal** [appsmithorg/appsmith#33312]
One of the feedback we received was around performance of drag and drop  of Building blocks.
Here is a list of ideas to try out: 
- [x] Add instrumentation to the backend flow to figure out optimizations 
                Added #33081 @AnaghHegde 
- [x] Cache the building blocks at server. #33440
- [ ] Not to save loading skeleton to the backend @jacquesikot . i.e. Saving and removing loader from the frontend only #33606
- [ ] `/block` API to return everything related to BB so that we don't have to make fetch calls after drop of BB. @AnaghHegde 
                 This will not have an impact on perceived performance of the drop behavior but still an optimisation.

**Comment by AnaghHegde** [comment]
[ ] RTS taking lot of time to calculate the bindings, need to debug the root cause for this slowness.

**Comment by AnaghHegde** [comment]
Another issue https://github.com/appsmithorg/appsmith/issues/33440

**Comment by rahulbarwal** [comment]
In recent weeks, we added two major performance boosters for the drag and drop of BBs: 
1. Added caching for templates at the platform server level. https://github.com/appsmithorg/appsmith/pull/33439
2. Fixing a bug that was causing multiple refactor calls to the RTS server https://github.com/appsmithorg/appsmith/pull/33439

This has brought down the numbers to 300ms to 1.5 secs range.
The last two ideas in description are still doable but we don't expect much of a performance boost from them. We will close this issue for now and reopen if performance creates problem for our users.
